aptos-move/aptos-vm/src/sharded_block_executor/sharded_aggregator_service.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::sharded_block_executor::aggr_overridden_state_view::TOTAL_SUPPLY_AGGR_BASE_VAL;
+use aptos_state_view::StateView;
+use aptos_types::{
+    state_store::state_key::StateKey, transaction::TransactionOutput,
+    write_set::TOTAL_SUPPLY_STATE_KEY,
+};
+use rayon::prelude::*;
+use serde::de::DeserializeOwned;
+use std::{ops, sync::Arc};
+
+pub fn get_state_value<S: StateView, T: DeserializeOwned>(
+    state_key: &StateKey,
+    state_view: &S,
+) -> Option<T> {
+    let value = state_view
+        .get_state_value_bytes(state_key)
+        .ok()?
+        .map(move |value| bcs::from_bytes(value.as_slice()));
+    value.transpose().map_err(anyhow::Error::msg).unwrap()
+}
+
+/// This class ensures that deltas can use all 128 bits without having to let go of the sign bit for
+/// cases where the delta is negative. That is, we don't have to use conversions to i128.
+/// However, it does not handle overflow and underflow. That is, it will indicate to the caller of
+/// the faulty logic with their usage of deltas.
+#[derive(Clone, Copy)]
+struct DeltaU128 {
```

Comment:
We have almost identical thing in aggregators.

https://github.com/aptos-labs/aptos-core/pull/9368/files#diff-87ad56dd16ac7e11a72f7798a0f39b4428b59c7090b93eafe2b4ee98e13d02e7


We should consolidate. I suggest naming it SignedU128, with signature:

pub enum SignedU128 {
    Positive(u128),
    Negative(u128),
}

and appropriate methods returning BoundedMathResult<SignedU128> to indicate overflow and underflow, which you can call expect() on if you want to throw.

cc @vusirikala

Reply:
Igor, yeah we should consolidate. Can I merge this change now though, and then consolidate it after the other PR is merged ? (I see that the other PR is still being worked upon)